core/objects/src/main/kotlin/app/aaps/core/objects/wizard/BolusWizard.kt

```diff
@@ -536,13 +545,18 @@ class BolusWizard @Inject constructor(
                                 ValueWithUnit.Gram(this@BolusWizard.carbs).takeIf { this@BolusWizard.carbs != 0 },
                                 ValueWithUnit.Minute(carbTime).takeIf { carbTime != 0 })
                         )
-                        commandQueue.bolus(this, object : Callback() {
-                            override fun run() {
-                                if (!result.success) {
-                                    uiInteraction.runAlarm(result.comment, rh.gs(app.aaps.core.ui.R.string.treatmentdeliveryerror), app.aaps.core.ui.R.raw.boluserror)
+                        var phoneNumber = preferences.get(StringKey.SmsReceiverNumber)
+                        if (preferences.get(BooleanKey.SmsAllowRemoteCommands) && !phoneNumber.isNullOrBlank()) {
```

Comment:
I will try to encapsulate it somehow

Reply:
I give up, I don't have time to resolve all the issues with gradle and dependencies during build